test/src/unit-capi-rest-dense_array.cc

```diff
@@ -302,7 +302,7 @@ DenseArrayRESTFx::~DenseArrayRESTFx() {
   config.set("rest.password", rest_server_password_);
 
   tiledb::sm::RestClient rest_client;
-  REQUIRE(rest_client.init(&config).ok());
```

Comment:
Good catch -- while this now compiles, it should segfault when run. We need to create a threadpool here:
```
ThreadPool tp;
REQUIRE(tp.init(4).ok());
REQUIRE(rest_client.init(&config, &tp).ok());
```